### PR TITLE
Remove dep upper pins

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?
We're moving to allow a wider set of dependencies. This gives the power to the user to specify the versions they need without getting in the way. See [this article](https://iscinumpy.dev/post/bound-version-constraints/) with the referenced links for justification.

### How was it fixed?
Removed all upper pins on dependencies that won't break. 

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -75,22 +75,22 @@
...
     install_requires=[
-        "aiohttp>=3.7.4.post0,<4",
-        "eth-abi>=3.0.0,<4.0.0",
-        "eth-account>=0.7.0,<0.8.0",
-        "eth-hash[pycryptodome]>=0.2.0,<1.0.0",
-        "eth-typing>=3.0.0,<4.0.0",
-        "eth-utils>=2.0.0,<3.0.0",
-        "hexbytes>=0.1.0,<1.0.0",
+        "aiohttp>=3.7.4.post0",
+        "eth-abi>=3.0.0",
```
Comment (fselmo) [inline setup.py:79]: We can probably stick this to `3.0.1` already maybe?

Comment (kclowes) [reply setup.py:79]: Since `web3.py` doesn't use any new features in `eth-abi 3.0.1` I think it might be better to leave it as open as possible.

Comment (fselmo) [reply setup.py:79]: sgtm 👍🏼

**Review by pacrob — APPROVED** [review]
lgtm!

**Pull request merged**